spec/memoizer_spec.rb

```diff
@@ -41,6 +84,96 @@ class Beepbop < MemoizerSpecClass; end
+      end
+
+      it "does not confuse one set of inputs for another" do
+        Timecop.freeze(today)
+        expect(object.with_hash_parameter(3)).to eq(today + 3)
+        expect(object.with_hash_parameter(3, subtract: true)).to eq(today - 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_hash_parameter(3)).to eq(today + 3)
+        expect(object.with_hash_parameter(3, subtract: true)).to eq(today - 3)
+      end
+    end
+
+    context "for a method with only a hash arg" do
+      it "stores memoized value" do
+        Timecop.freeze(today)
+        expect(object.with_only_hash_parameter(ndays: 3)).to eq(today + 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_only_hash_parameter(ndays: 3)).to eq(today + 3)
+      end
+
+      it "does not confuse one set of inputs for another" do
+        Timecop.freeze(today)
+        expect(object.with_only_hash_parameter(ndays: 3)).to eq(today + 3)
+        expect(object.with_only_hash_parameter(ndays: 3, subtract: true)).to eq(today - 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_only_hash_parameter(ndays: 3)).to eq(today + 3)
+        expect(object.with_only_hash_parameter(ndays: 3, subtract: true)).to eq(today - 3)
+      end
+    end
+
+    context "for a method with a mix of positional and keyword arguments" do
+      it "stores memoized value" do
+        Timecop.freeze(today)
+        expect(object.with_kwargs(3)).to eq(today + 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_kwargs(3)).to eq(today + 3)
+      end
+
+      if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0')
+        it 'raises an error when the keyword args are passed as a hash' do
+          expect { object.with_kwargs(3, { subtract: true }) }.to raise_error(
+            ArgumentError,
+            'wrong number of arguments (given 2, expected 1)'
+          )
+        end
+      end
+
+      it "does not confuse one set of inputs for another" do
+        Timecop.freeze(today)
+        expect(object.with_kwargs(3)).to eq(today + 3)
+        expect(object.with_kwargs(3, **{ subtract: true })).to eq(today - 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_kwargs(3)).to eq(today + 3)
+        expect(object.with_kwargs(3, **{ subtract: true })).to eq(today - 3)
+      end
+    end
+
+    context "for a method with only keyword args" do
+      it "stores memoized value" do
+        Timecop.freeze(today)
+        expect(object.with_only_kwargs(ndays: 3)).to eq(today + 3)
+        Timecop.freeze(tomorrow)
+        expect(object.with_only_kwargs(ndays: 3)).to eq(today + 3)
+      end
+
+      if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0')
+        it 'raises an error when the keyword args are passed as a hash' do
+          expect { object.with_only_kwargs({ ndays: 3, subtract: true }) }.to raise_error(
+              ArgumentError,
+              'wrong number of arguments (given 2, expected 1)'
```

Comment:
And now I'm confused about how 2 arguments are provided, when they're passed as a hash.

Reply:
@jchuang I didn't run this locally, and it failed on CI. 🙂

The expected error message should be `wrong number of arguments (given 1, expected 0; required keyword: ndays)`.